compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -442,21 +447,37 @@ static BOOL_TYS: TyS<'static> = TyS {
     kind: ty::Bool,
     flags: TypeFlags::empty(),
     outer_exclusive_binder: DebruijnIndex::from_usize(0),
+    stable_hash: Fingerprint::ZERO,
 };
 
 impl<'a, 'tcx> HashStable<StableHashingContext<'a>> for Ty<'tcx> {
     fn hash_stable(&self, hcx: &mut StableHashingContext<'a>, hasher: &mut StableHasher) {
```

Comment:
This does not verify that the `hcx` has the same settings as the one used for caching. We somehow need to verify that.

Reply:
while I don't know of any `Span` or `HirId` within `Ty<'tcx>`, I added a reset of the settings to the default settings as a safety net.